tmt/steps/discover/__init__.py

```diff
@@ -111,8 +111,8 @@ def extract_distgit_source(
             with tmt.utils.retry_session() as session:
                 response = session.get(url)
             response.raise_for_status()
-            os.makedirs(target_dir, exist_ok=True)
-            with open(os.path.join(target_dir, source_name), 'wb') as tarball:
+            target_dir.mkdir(exist_ok=True)
```

Comment:
```suggestion
            target_dir.mkdir(parents=True, exist_ok=True)
```

We should ensure any intermediate directories are created if needed as well, right?

Reply:
Correct, fixed.

Reply:
Good.